### PR TITLE
Timer high fps bugs

### PR DESCRIPTION
**Pull request by griseraner**

### CODE REVIEW

File: Code/CryGame/Items/Weapons/FireModes/Plant.cpp [hunk 1]
```diff
@@ -314,17 +311,19 @@ void CPlant::StartFire()
...
 	m_planting = true;
 	m_pWeapon->SetBusy(true);
 	m_pWeapon->PlayAction(ItemString(m_plantactions.plant.c_str()));
 
-	m_plantTimer = m_plantparams.delay;
+	m_pWeapon->GetScheduler()->TimerAction(m_pWeapon->GetCurrentAnimationTime(CItem::eIGS_FirstPerson), 
+		CSchedulerAction<StartPlantAction>::Create(this), false);
 
-	m_pWeapon->GetScheduler()->TimerAction(m_pWeapon->GetCurrentAnimationTime(CItem::eIGS_FirstPerson), CSchedulerAction<StartPlantAction>::Create(this), false);
+	m_pWeapon->GetScheduler()->TimerAction(static_cast<uint32>(m_plantparams.delay * 1000.f), //CryMP added
```
Comment (ccomrade) [inline Code/CryGame/Items/Weapons/FireModes/Plant.cpp:325]:
```suggestion
	m_pWeapon->GetScheduler()->TimerAction(static_cast<uint32_t>(m_plantparams.delay * 1000.f), //CryMP added
```

**Review by ccomrade — APPROVED** [review]

File: Code/CryGame/Items/Weapons/FireModes/Scan.cpp [hunk 2]
```diff
@@ -93,97 +91,105 @@ void CScan::Activate(bool activate)
...
 		{
-			if (m_tagEntitiesDelay > 0.0f)
-			{
-				m_tagEntitiesDelay -= frameTime;
-				if (m_tagEntitiesDelay <= 0.0f)
-				{
-					m_tagEntitiesDelay = 0.0f;
+			pSound->SetLoopMode(true);
+		}
+		pScan->CallTimers();
```
Comment (griseraner) [inline Code/CryGame/Items/Weapons/FireModes/Scan.cpp:115]: This is ok?

Comment (ccomrade) [inline Code/CryGame/Items/Weapons/FireModes/Scan.cpp:115]:
```suggestion
		pScan->StartTimers();
```

Comment (ccomrade) [reply Code/CryGame/Items/Weapons/FireModes/Scan.cpp:115]: Better name needed.

File: Code/CryGame/Items/Weapons/FireModes/Scan.cpp [hunk 3]
```diff
@@ -93,97 +91,105 @@ void CScan::Activate(bool activate)
...
 
-		m_pWeapon->RequireUpdate(eIUS_FireMode);
+	void execute(CItem* _this)
+	{
+		pScan->StopFire();
 	}
+};
+
+//------------------------------------------------------------------------
+void CScan::CallTimers()
```
Comment (ccomrade) [inline Code/CryGame/Items/Weapons/FireModes/Scan.cpp:153]:
```suggestion
void CScan::StartTimers()
```

File: Code/CryGame/Items/Weapons/FireModes/Scan.h [hunk 4]
```diff
@@ -42,6 +46,8 @@ class CScan : public IFireMode
 
 	virtual void Activate(bool activate);
 
+	void CallTimers();
```
Comment (ccomrade) [inline Code/CryGame/Items/Weapons/FireModes/Scan.h:49]:
```suggestion
	void StartTimers();
```

**Pull request merged**